### PR TITLE
Wh exclude window support

### PR DESCRIPTION
**Pull request by gane5hvarma**
## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

---
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-server/647)

### CODE REVIEW

**Review by sivashanmukh — APPROVED** [review]
LGTM

File: warehouse/utils/utils.go [hunk 1]
```diff
@@ -470,6 +481,19 @@ func GetConfigValueBoolString(key string, warehouse WarehouseT) string {
 	return "false"
 }
 
+func GetConfigValueAsMap(key string, warehouse WarehouseT) map[string]interface{} {
+	config := warehouse.Destination.Config
+	value := map[string]interface{}{}
+	if config[key] != nil {
+		fmt.Println(config[key])
```
Comment (SumanthPuram) [inline warehouse/utils/utils.go:488]: Remove this.

Comment (SumanthPuram) [inline warehouse/utils/utils.go:488]: Remove this.

File: warehouse/utils/utils.go [hunk 2]
```diff
@@ -470,6 +481,19 @@ func GetConfigValueBoolString(key string, warehouse WarehouseT) string {
...
+		if val, ok := config[key].(map[string]interface{}); ok {
+			return val
+		}
+		fmt.Println("cant convert")
```
Comment (SumanthPuram) [inline warehouse/utils/utils.go:492]: Remove this.

File: warehouse/utils/utils.go [hunk 3]
```diff
@@ -47,12 +47,23 @@ const (
 )
 
 const (
+	UploadStatusField          = "status"
```
Comment (SumanthPuram) [inline warehouse/utils/utils.go:50]: Are these changes part of this PR?

File: warehouse/scheduling_test.go [hunk 4]
```diff
@@ -54,6 +54,39 @@ var _ = Describe("Warehouse", func() {
...
+				Expect(CheckCurrentTimeExistsInExcludeWindow(currentTime, startTime, endTime)).To(Equal(false))
+				startTime = "22:00"
+				endTime = "06:00"
+				currentTime = time.Date(2009, time.November, 10, 21, 30, 0, 0, time.UTC)
+				Expect(CheckCurrentTimeExistsInExcludeWindow(currentTime, startTime, endTime)).To(Equal(false))
+				startTime = ""
+				endTime = ""
+				currentTime = time.Date(2009, time.November, 10, 21, 30, 0, 0, time.UTC)
+				Expect(CheckCurrentTimeExistsInExcludeWindow(currentTime, startTime, endTime)).To(Equal(false))
+			})
```
Comment (SumanthPuram) [inline warehouse/scheduling_test.go:89]: Please add a test case, where one of them is empty.

File: warehouse/utils/utils.go [hunk 5]
```diff
@@ -470,6 +481,19 @@ func GetConfigValueBoolString(key string, warehouse WarehouseT) string {
 	return "false"
 }
 
+func GetConfigValueAsMap(key string, warehouse WarehouseT) map[string]interface{} {
```
Comment (SumanthPuram) [inline warehouse/utils/utils.go:484]: Can we change the function signature to take destinationConfig instead of warehouse?

```
func GetConfigValueAsMap(key string, destConfig map[string]interface{} ) map[string]interface{} 
```

**Review by SumanthPuram — APPROVED** [review]

File: warehouse/utils/utils.go (hunk 3 above)
Comment (gane5hvarma) [reply warehouse/utils/utils.go:50]: no, it got added during merge

**Review by SumanthPuram — APPROVED** [review]
